components/raftstore/src/store/peer.rs

```diff
@@ -1796,6 +1806,19 @@ where
         );
         RAFT_READ_INDEX_PENDING_COUNT.sub(read.cmds.len() as i64);
         for (req, cb, mut read_index) in read.cmds.drain(..) {
+            // leader reports key is locked
+            if let Some(locked) = read.locked.take() {
```

Comment:
Do we need this check on leader?

Reply:
Both ReadIndex RPC and replica read need to get the lock from the request. So we need to extract it.